metrics-operator/controllers/analysis/objectives_evaluator.go

```diff
@@ -0,0 +1,43 @@
+package analysis
+
+import (
+	"github.com/go-logr/logr"
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	metricstypes "github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/analysis/types"
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/providers"
+	"golang.org/x/net/context"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate moq -pkg fake -skip-ensure -out ./fake/evaluator_mock.go . IObjectivesEvaluator
+type IObjectivesEvaluator interface {
+	Evaluate(ctx context.Context, providerType string, obj chan metricstypes.ProviderRequest)
+}
+
+type ObjectivesEvaluator struct {
+	*metricsapi.Analysis
+	providers.ProviderFactory
+	client.Client
+	Log     logr.Logger
+	results chan metricstypes.ProviderResult
+}
+
+func (oe ObjectivesEvaluator) Evaluate(ctx context.Context, providerType string, obj chan metricstypes.ProviderRequest) {
+	provider, err := oe.ProviderFactory(providerType, oe.Log, oe.Client)
+	if err != nil {
+		oe.Log.Error(err, "Failed to get the correct Provider")
+	}
+	for o := range obj {
+		value := ""
+		if err == nil {
+			value, _, err = provider.FetchAnalysisValue(ctx, o.Query, oe.Analysis.Spec, o.Provider)
+		}
```

Comment:
This does not make much sense, can you please explain your thoughts here ?

Reply:
this part is in the TODO and needs discussion, I am injecting here the error coming from the provider part... we need to rethink when we do stop the reconciler and when we store errors

Reply:
but indeed should be o.err :D

Reply:
I tought there is no need to compute the value if we do have an error from the provider creation, but we may still want to move on with the analysis so we return the error for all the obj under that provider

Reply:
this is part of the TODO where I mention when should we error and stop fail and when should we forward this error and store it in the analysis